From da26dfbf78468f5ed724e022b300a07862a95833 Mon Sep 17 00:00:00 2001
From: Diego Escalante Urrelo <diegoe@gnome.org>
Date: Sun, 27 Oct 2019 03:45:20 -0500
Subject: [PATCH] build: Update gtk-doc generation to meson changes

After updating our use of link_with/link_whole for meson-0.52.0 we
accidentally broke gtk-doc generation.

Before c8b862b491cfbbb4f79b24d7cd90e4fb1f37cb9f gnome_builder_deps
included link_whole calls to all our libide static libraries. Because we
changed those calls to link_with, libide_gtk_doc was no longer linking
properly.

To solve the above we now use gnome_builder_static, which is a list of
all our internal libide static libraries and can be used as an argument
to link_whole, just like we do in the gnome_builder target.

Fixes: https://gitlab.gnome.org/GNOME/gnome-builder/issues/1063
---
 doc/sdk/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/sdk/meson.build b/doc/sdk/meson.build
index 4bb2421b3..f6ff1a3b7 100644
--- a/doc/sdk/meson.build
+++ b/doc/sdk/meson.build
@@ -30,7 +30,7 @@ docpath = join_paths(get_option('datadir'), 'gtk-doc', 'html')
 
 libide_gtk_doc = shared_library('ide-gtk-doc',
             c_args: libide_args + release_args,
-      dependencies: gnome_builder_deps,
+        link_whole: gnome_builder_static,
 )
 
 libide_gtk_doc_dep = declare_dependency(
-- 
2.24.1

